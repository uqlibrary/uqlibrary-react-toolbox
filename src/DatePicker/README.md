# TextField

A simple DatePicker component that lists items for UQ Library applications using ReactJs

## Props
- `standard props from http://www.material-ui.com/#/components/date-picker`
- `helpTitle: PropTypes.string`
- `helpText: PropTypes.any`


## Usage

**./App.js**
```jsx
import React from 'react';
...

import {DatePicker} from 'uqlibrary-react-toolbox';
 
const App = () => (
  <div>
    <DatePicker 
        name="sampleFieldName" 
        fullWidth />
  </div>
);

ReactDOM.render(
  <App />,
  document.getElementById('app')
);
```
