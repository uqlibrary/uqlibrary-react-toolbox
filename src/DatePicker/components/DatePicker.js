import React from 'react';
import PropTypes from 'prop-types';
import DatePicker from 'material-ui/DatePicker';

import propFilter from '../../helpers/_filterProps';
import {HelpIcon} from '../../HelpDrawer';

export default function DatePickerWrapper(props) {
    const filteredProps = propFilter(props, DatePicker.propTypes);
    filteredProps.name = props.input ? props.input.name : props.name;
    filteredProps.floatingLabelText = props.floatingLabelText;
    filteredProps.floatingLabelFixed = props.floatingLabelFixed;

    filteredProps.value = filteredProps.value instanceof Date ? filteredProps.value : new Date();
    filteredProps.onChange = (e, value) => props.input.onChange(value);
    delete filteredProps.onBlur;

    return (
        <div style={{position: 'relative', width: '100%'}}>
            <DatePicker {...filteredProps} />
            {props.helpText && (
                <div style={{position: 'absolute', top: '20px', right: 0}}>
                    <HelpIcon title={props.helpTitle} text={props.helpText} buttonLabel="Got it!" />
                </div>
            )}
        </div>
    );
}

DatePickerWrapper.propTypes = {
    ...DatePicker.propTypes,
    helpTitle: PropTypes.string,
    helpText: PropTypes.any
};
