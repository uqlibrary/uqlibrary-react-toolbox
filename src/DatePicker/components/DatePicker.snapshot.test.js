jest.dontMock('./DatePicker');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import DatePicker from './DatePicker';

function setup(props) {
    return shallow(<DatePicker {...props} />);
}

describe('DatePickerWrapper snapshots tests', () => {
    it('renders DatePicker component', () => {
        const props =
            {
                name: 'test',
                type: 'text',
                fullWidth: true,
                floatingLabelText: 'This is a test date picker component'
            };

        const app = setup(props);
        expect(toJson(app)).toMatchSnapshot();
    });
});
