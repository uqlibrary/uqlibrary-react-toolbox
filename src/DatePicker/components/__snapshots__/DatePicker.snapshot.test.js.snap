// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePickerWrapper snapshots tests renders DatePicker component 1`] = `
<div
  style={
    Object {
      "position": "relative",
      "width": "100%",
    }
  }
>
  <DatePicker
    autoOk={false}
    container="dialog"
    disableYearSelection={false}
    disabled={false}
    firstDayOfWeek={1}
    floatingLabelText="This is a test date picker component"
    fullWidth={true}
    hideCalendarDate={false}
    name="test"
    onChange={[Function]}
    style={Object {}}
    type="text"
    value={2017-07-14T00:14:02.314Z}
  />
</div>
`;
